internal/security/config/command/proxy/tls/command.go

```diff
@@ -61,9 +78,208 @@ func NewCommand(
 }
 
 func (c *cmd) Execute() (statusCode int, err error) {
-	fmt.Println("TODO: Configure inbound TLS certificate.")
-	fmt.Printf("--incert %s\n", c.certificatePath)
-	fmt.Printf("--inkey %s\n", c.privateKeyPath)
-	err = fmt.Errorf("tls command is unimplemented")
-	return interfaces.StatusCodeExitWithError, err
+
+	if err := c.uploadProxyTlsCert(); err != nil {
+		return interfaces.StatusCodeExitWithError, err
+	}
+
+	return
+}
+
+func (c *cmd) readCertKeyPairFromFiles() (*bootstrapConfig.CertKeyPair, error) {
+	certPem, err := ioutil.ReadFile(c.certificatePath)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read TLS certificate from file %s: %w", c.certificatePath, err)
+	}
+	prvKey, err := ioutil.ReadFile(c.privateKeyPath)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read private key from file %s: %w", c.privateKeyPath, err)
+	}
+	return &bootstrapConfig.CertKeyPair{Cert: string(certPem), Key: string(prvKey)}, nil
+}
+
+func (c *cmd) uploadProxyTlsCert() error {
+	// try to read both files and make sure they are existing
+	certKeyPair, err := c.readCertKeyPairFromFiles()
+	if err != nil {
+		return err
+	}
+
+	// to see if any proxy certificates already exists
+	// if yes, then delete them all first
+	// and then upload the new TLS certificate
+	existingCertIDs, err := c.listKongTLSCertificates()
+	if err != nil {
+		return err
+	}
+
+	c.loggingClient.Debug(fmt.Sprintf("number of existing Kong tls certs = %d", len(existingCertIDs)))
+
+	if len(existingCertIDs) > 0 {
+		// delete the existing certs first
+		// ideally, it should only have one certificate to be deleted
+		// Disclaimer: Kong TLS certificate should only be uploaded via secret-config utility
+		for _, certID := range existingCertIDs {
+			if err := c.deleteKongTLSCertificateById(certID); err != nil {
+				return err
+			}
+		}
+	}
+
+	// post the certKeyPair as the new one
+	if err := c.postKongTLSCertificate(certKeyPair); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *cmd) listKongTLSCertificates() ([]string, error) {
+	// list certificates first to see if any already exists
+	certKongURL := strings.Join([]string{c.configuration.KongURL.GetProxyBaseURL(), "certificates"}, "/")
```

Comment:
I have decided that this algorithm is unsafe.  As Kong uses a central "certificates" table for all certificates no matter what their purpose is, I think it is better to enumerate the /snis  API, get the unique list of certificate IDs, and delete those instead.  A blanket deletion of certificates might get certificates that aren't being used to serve TLS.

Reply:
ok

Reply:
changed and now it uses `/snis` to figure out the unique certificate ids to be deleted.